### PR TITLE
Feature >> First working check

### PR DESCRIPTION
**Pull request by jbaldrich**
This is just an example of a fully functional check to demonstrate how I should build the logic.

Please, note that JSON still needs to be populated through a more efficient method.

### CODE REVIEW

**Review by polansk1 — APPROVED** [review]

File: admin/class-wp-security-bp-admin.php [hunk 1]
```diff
@@ -216,6 +216,23 @@ public function fix_wp_config() {
...
+	 * This function fires the fix wp-config.php feature
+	 *
+	 * @since    1.0.0
+	 */
+	public function fix_stuff() {
+
+		if ( $_POST['do-shit'] === 'file-shit' ) {
+			$files = new WP_Security_BP_Files( $this->plugin_name, $this->admin_url );
+			$files->move_wp_config();
+			$this->final_json();
```
Comment (polansk1) [inline admin/class-wp-security-bp-admin.php]: I think we have to find the way to update the check alone instead call `final_json()` to update all checks. For now it's ok to test.

File: admin/class-wp-security-bp-admin.php [hunk 2]
```diff
@@ -169,23 +185,83 @@ public function add_action_links( $links ) {
...
+		
+		wp_send_json( $response );
+
+	}
+
+	/**
+	 * This function gets the action sent through POST and calls the appropiate method.
+	 *
+	 * @since    1.0.0
+	 */
```
Comment (uri-ustrell) [inline admin/class-wp-security-bp-admin.php:214]: We could add properties initialized here. Some coding App intellisense use block comment to provide hints.

File: admin/class-wp-security-bp-admin.php [hunk 3]
```diff
@@ -169,23 +185,83 @@ public function add_action_links( $links ) {
...
+					default:
+						wp_die();
+				}
+				
+				$class->$method();
+				$this->check_all();
+				
+			}
+
+		}
```
Comment (uri-ustrell) [inline admin/class-wp-security-bp-admin.php:263]: There is a Lack of alternative. what to do if $action is not in $actions. It's in every check we do in this class almost.

In fact, to be fair, there is a lack of alternatives at plugin wide level.(what to do if not the expected parameter or iteration)

we must start refactor this.


There is nothing to do with this part of the code if variable doesn't match the actual cardcoded array. OK. But we don't pay attention at next. I mean , the rest of the code stills there and logic happens in one way or another. 

As this is an open source development, we must take even more care in this details. 

🔐

**Review by polansk1 — APPROVED** [review]

File: admin/class-wp-security-bp-admin.php (hunk 3 above)
Comment (jbaldrich) [reply admin/class-wp-security-bp-admin.php:263]: Totally agree 👍

**Pull request merged**